docs/modules/firewall.rst

```diff
@@ -12,124 +12,143 @@ firewall
 
-    inbound (optional, list, None)
-      A list of rules for inbound traffic.
 
+          label (True, str, None)
+          The label of this rule
 
-      label (True, str, None)
-        The label of this rule.
 
+          ports (False, str, None)
+          A string representing the port or ports on which traffic will be allowed.
 
-      action (True, str, None)
-        Controls whether traffic is accepted or dropped by this rule.
+          See https://www.linode.com/docs/api/networking/#firewall-create
 
 
-      description (optional, str, None)
-        The description of this rule.
+          protocol (False, str, None)
+          The type of network traffic to allow.
 
 
-      addresses (optional, list, None)
-        Allowed IPv4 or IPv6 addresses.
 
+      inbound_policy (False, str, None)
+       The default behavior for inbound traffic.
 
-        ipv4 (optional, list, None)
-          A list of IPv4 addresses or networks.
 
-          Must be in IP/mask format.
+      outbound (False, list, None)
+       A list of rules for outbound traffic.
 
 
-        ipv6 (optional, list, None)
-          A list of IPv4 addresses or networks.
+          action (True, str, None)
+          Controls whether traffic is accepted or dropped by this rule.
 
-          Must be in IP/mask format.
 
+          addresses (False, dict, None)
+          Allowed IPv4 or IPv6 addresses.
 
 
+              ipv4 (False, list, None)
+             A list of IPv4 addresses or networks.
 
-    outbound (optional, list, None)
-      A list of rules for outbound traffic.
+             Must be in IP/mask format.
 
 
-      label (True, str, None)
-        The label of this rule.
+              ipv6 (False, list, None)
+             A list of IPv4 addresses or networks.
 
+             Must be in IP/mask format.
 
-      action (True, str, None)
-        Controls whether traffic is accepted or dropped by this rule.
 
 
-      description (optional, str, None)
-        The description of this rule.
+          description (False, str, None)
+          A description for this rule.
 
 
-      addresses (optional, list, None)
-        Allowed IPv4 or IPv6 addresses.
+          label (True, str, None)
+          The label of this rule
 
 
-        ipv4 (optional, list, None)
-          A list of IPv4 addresses or networks.
+          ports (False, str, None)
+          A string representing the port or ports on which traffic will be allowed.
 
-          Must be in IP/mask format.
+          See https://www.linode.com/docs/api/networking/#firewall-create
 
 
-        ipv6 (optional, list, None)
-          A list of IPv4 addresses or networks.
+          protocol (False, str, None)
+          The type of network traffic to allow.
 
-          Must be in IP/mask format.
 
 
+      outbound_policy (False, str, None)
+       The default behavior for outbound traffic.
 
 
 
+  status (False, str, None)
+    The status of this Firewall.
+
+
```

Comment:
Is the indentation correct here? ... also, the extra trailing newlines

Reply:
The trailing newlines are unfortunately unavoidable due to the way we handle templates :disappointed: